docs/current_docs/features/shell.mdx

```diff
@@ -67,13 +71,30 @@ container |
 
 ![Dagger Shell](/img/current_docs/features/shell-build.gif)
 
-:::note
-Dagger Shell allows you to execute Dagger Functions as native shell commands, using a Bash-compatible syntax. However, instead of calling a subprocess for execution, Dagger Shell directly interprets commands within the Dagger CLI and executes them in containers. This means that all your commands execute as API calls in a secure, sandboxed environment, rather than as processes with direct access to the host system.
-:::
+## Bash frontend, Dagger backend
+
+Dagger Shell uses the Bash syntax as a frontend, but its behavior is quite different in the backend:
+
+- Instead of executing unix commands, you execute Dagger Functions
+- Instead of passing text streams from command to command, you pass typed objects from function to function
+- Instead of available commands being the same everywhere in the pipeline, each command is executed in the context of the previous command's output object. For example, `foo | bar` really means `foo().bar()`
+- Instead of using the local host as an execution environment, you use containerized runtimes
+- Instead of being mixed with regular commands, shell builtins are prefixed with `.` (similar to sqlite)
 
-## Modes of execution
+Besides these differences, all the features of the Bash *syntax* are available in Dagger Shell, including:
 
-Dagger Shell supports multiple ways to run commands:
+- Shell variables: `container=$(container | from alpine)`
+- Shell functions: `container() { container | from alpine; }`
+- Job control: `frontend | test & backend | test & wait`
+- Quoting: single quotes and double quotes have the same meaning as in bash
+
+## Tracing
+
+Dagger supports opentelemetry tracing out of the box, and Dagger Shell takes advant
```

Comment:
Quick note: incomplete sentence

Reply:
Thank you! My kids startled me and I lost track when I resumed work.